wandb/sdk/internal/internal_api.py

```diff
@@ -98,13 +111,20 @@ def __init__(
             )
         )
         self.retry_callback = retry_callback
-        self.gql = retry.Retry(
-            self.execute,
-            retry_timedelta=retry_timedelta,
-            check_retry_fn=util.no_retry_auth,
-            retryable_exceptions=(RetryError, requests.RequestException),
-            retry_callback=retry_callback,
-        )
+        try:
+            self.gql = retry.Retry(
+                self.execute,
+                retry_timedelta=retry_timedelta,
+                check_retry_fn=util.no_retry_auth,
+                retryable_exceptions=(RetryError, requests.RequestException,),
+                retry_warning=local_instance_warning,
+                retry_callback=retry_callback,
+            )
+        except Exception as e:
+            if local_instance_warning:
+                raise Exception(local_instance_warning)
+            else:
+                raise e
```

Comment:
What is the `try` -> `except` for?  Won't the retrier keep retrying for hours and just print a warning?

Reply:
Its for the JSONDecodeError (see  jira ticket)

Reply:
The recent retry fixes should ensure we don't raise exceptions when content isn't JSON encoded so the retry logic will retry it forever and this except case will only be reached after we stop retrying...